internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -467,6 +479,114 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *adiiov1alpha1.Agent
 	return nil, nil
 }
 
+func (r *BMACReconciler) isDay2Worker(agent *adiiov1alpha1.Agent) bool {
+	return agent.Spec.Role == models.HostRoleWorker && agent.Spec.ClusterDeploymentName != nil
+}
+
+func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, agent *adiiov1alpha1.Agent) reconcileResult {
+	cdKey := types.NamespacedName{
+		Namespace: agent.Spec.ClusterDeploymentName.Namespace,
+		Name:      agent.Spec.ClusterDeploymentName.Name,
+	}
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	if err := r.Get(ctx, cdKey, clusterDeployment); err != nil {
+		r.Log.WithError(err).Errorf("failed to get clusterDeployment resource %s/%s", cdKey.Namespace, cdKey.Name)
+		return reconcileError{err}
+	}
+
+	// Secret contains kubeconfig for the spoke cluster
+	secret := &corev1.Secret{}
+	name := fmt.Sprintf(adminKubeConfigStringTemplate, clusterDeployment.Name)
+	err := r.Get(ctx, types.NamespacedName{Namespace: clusterDeployment.Namespace, Name: name}, secret)
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.WithError(err).Errorf("failed to get secret resource %s/%s", clusterDeployment.Namespace, name)
+		// If secret is not found, wait until a reconcile is trigged by a watch event
+		return reconcileComplete{}
+	} else if err != nil {
+		return reconcileError{err}
+	}
+
+	spokeClient, err := getSpokeClient(secret)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to create spoke kubeclient")
+		return reconcileError{err}
+	}
+
+	machine, err := r.createSpokeMachine(ctx, spokeClient, bmh, clusterDeployment)
+	if err != nil {
+		r.Log.WithError(err).Errorf("failed to create or update spoke Machine")
+		return reconcileError{err}
+	}
+
+	bmhSpoke := &bmh_v1alpha1.BareMetalHost{}
+	err = spokeClient.Get(ctx, types.NamespacedName{Name: bmh.Name, Namespace: bmh.Namespace}, bmhSpoke)
+	if err != nil && errors.IsNotFound(err) {
+		bmhSpoke = &bmh_v1alpha1.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      bmh.Name,
+				Namespace: bmh.Namespace,
+			},
+		}
+		bmhSpoke.Spec = bmh.Spec
```

Comment:
we need to add HW details as annotations as it is described here https://github.com/metal3-io/baremetal-operator/blob/master/docs/inspectAnnotation.md

Reply:
I think we should add the HW details to hub BMH and just copy the spec over to spoke BMH. WDYT?

Reply:
i think HW details are part of status and in any case better to give spoke ironic to handle this. Talked to @hardys about it. He said we should add it to annotation.

Reply:
Copied the annotations over from hub bmh.

Reply:
But HW will not be there. I think we should take HW from status or spec and push it to annotation on spoke with another one we need when we did it manually

Reply:
i am still not sure if we should set full spec from hub . Lets validate it with metal3 team

Reply:
@flaper87 @sacharya  i am wondering what will happened with HW annotation. Ironic will delete this annotations after setting vars to status. So i am not sure it is enough to copy them. What you think?

Reply:
> @flaper87 @sacharya i am wondering what will happened with HW annotation. Ironic will delete this annotations after setting vars to status. So i am not sure it is enough to copy them. What you think?

Not sure I follow, but I may be missing something about what Ironic does. We are already setting `inspect: disabled` so, AFAIU, Ironic should not touch any of this data. For the spoke cluster we will also set it to unmanaged (or whatever the new setting is). Correct?

Am I missing something?

Reply:
@flaper87 yep i think you miss ironic behavior.
when you set "inspect.metal3.io: disabled" it just says to ironic go and find HW inside annotation.
When it will find it there it will add them to status and remove the annotation.
@hardys i am right?
Will it leave "inspect.metal3.io: disabled" ?

Reply:
@tsorya yes that's correct, if you set the `inspect.metal3.io: disabled` annotation that will never be removed by the BMO, and the ironic inspection will be skipped.

https://github.com/metal3-io/baremetal-operator/blob/master/docs/inspectAnnotation.md

The `inspect.metal3.io/hardwaredetails` annotation is removed at the point where it's consumed by the BMO and the data is copied into the BMH status field (you can set it again with new data, but only if `inspect.metal3.io: disabled` is defined)

Reply:
> When it will find it there it will add them to status and remove the annotation.
> Will it leave "inspect.metal3.io: disabled" ?

As @hardys mentioned, it will remove the `hardwaredetails` annotation but not the `inspect` one.

We set `hardwaredetails` during the reconcile and `inspect.metal3.io: disabled` is disabled so we can assume that the data in the BMH is coming from the AI agent so it's aligned with what AI "knows" about the node.

BMAC was implemented based on the above assumptions. I think we are fine, aren't we?

Reply:
We will need to set HW details for spoke cluster the same way we doing for hub one. We can't rely on annotation on hub cause it can be deleted by ironic before we will copy it

Reply:
Updated. PTAL!